src/coroutine.js

```diff
@@ -1,25 +1,21 @@
+import Action from './Action'
+
 export default function (resolve, iterator, promise) {
 	new Coroutine(resolve, iterator, promise).run()
+	// taskQueue.add(new Coroutine(resolve, iterator, promise))
```

Comment:
After having <s>re-read</s> skimmed over the [`async`/`await` proposal](https://tc39.github.io/ecmascript-asyncawait/) it seems we should start immediately but then use `_when` instead of `_runAction` below in the `handle` method after the call.